### ISSUE TITLE
F-Droid

### ISSUE THREAD

**Issue by ghost** [dashevo/dash-wallet#19]
Load this app on official F-Droid repo for promote the development and dissemination, please!

**Comment by HashEngineering** [comment]
So far we haven't done anything on this request.

**Comment by herbsmn** [comment]
China would appreciate it. https://www.dash.org/forum/threads/dash-wallet-for-android-release-4-closed.4843/page-2#post-61213

**Comment by herbsmn** [comment]
The first step seems to be submitting a request for packaging: https://gitlab.com/fdroid/rfp

Do you have any questions about the process? Perhaps I could help figure out the answers.

Are there any known blockers that you can enumerate?

**Comment by herbsmn** [comment]
Here's a guide I found that describes the process: https://fdroid.gitlab.io/fdroid-website/docs/Submitting_to_F-Droid_Quick_Start_Guide/

**Comment by HashEngineering** [comment]
Currently installing the fdroid server, etc.

**Comment by HashEngineering** [comment]
its taking a long time.  My Hard Disk kept running out of space!  Work continues.

**Comment by herbsmn** [comment]
> its taking a long time.  My Hard Disk kept running out of space!  Work continues.

Thanks for your work making Dash Wallet more accessible!

Other than every Chinese android user, which has to be a huge amount of
people, another community that will benefit from the addition of the
dash wallet in F-Droid is the 'post-Snowden internet' community.

These are people that don't trust Prism Partner mega-corps, like Google,
and run android ROMs like https://LineageOS.org or
https://prism-break.org/en/projects/replicant/ without Google Play.

Your work is very much appreciated. Excited to hear how it progresses.

**Comment by HashEngineering** [comment]
The fdroid system was setup.  Now its time to build the app within their system.

**Comment by HashEngineering** [comment]
More progress made.  dashj has been added to the jCenter repository along with dash-java-wns-server.  These are two dependencies that are required by dash-wallet.  We are getting closer to building the app for fdroid.

At this point, we need to deal with the compiled libraries that are part of the app for the x11 hash (for faster hashing).  for this first version, we will simply remove them on fdroid.  Later we will determine how to compile them during the wallet build.

**Comment by HashEngineering** [comment]
Update - completed tasks from previous statements, submitted merge request to add Dash Wallet to Fdroid.  Will post again when it is part of the Fdroid app results.

**Comment by HashEngineering** [comment]
Process is slow.  Had to fix a few errors.

**Comment by herbsmn** [comment]
thanks for the updates and for your continued work on this!

**Comment by HashEngineering** [comment]
Things are moving slow.  According to the records of FDroid, the build of the app was successful.  However, I haven't seen the app published such that it can be downloaded.

https://f-droid.org/wiki/index.php?title=hashengineering.darkcoin.wallet/lastbuild_40056&redirect=no

**Comment by herbsmn** [comment]
strange. i wonder what's going on.

**Comment by herbsmn** [comment]
perhaps you forgot to file a Request For Packaging? https://gitlab.com/fdroid/rfp

It seems like you have all of the hard work done, you simply need to file an issue at that gitlab link and it should be accepted. nice job!

**Comment by herbsmn** [comment]
looks like you did a merge request here: https://gitlab.com/fdroid/fdroiddata/merge_requests
but you didn't file an issue here asking to be packaged: https://gitlab.com/fdroid/rfp/issues

**Comment by HashEngineering** [comment]
I thought that doing the merge request was enough and the RFP request was
for those who didn't do the merge request.

Nevertheless I will contact their support.

On Apr 13, 2017 12:30 AM, "FreeKurt.com" <notifications@github.com> wrote:

looks like you did a merge request here: https://gitlab.com/fdroid/
fdroiddata/merge_requests
but you didn't file an issue here asking to be packaged:
https://gitlab.com/fdroid/rfp/issues

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/HashEngineering/dash-wallet/issues/19#issuecomment-293777800>,
or mute the thread

**Comment by herbsmn** [comment]
how's it going?

**Comment by HashEngineering** [comment]
After contacting support we learned of a few issues that we need to resolve to get it to work.  We are working on those things now.

**Comment by HashEngineering** [comment]
There were several issues that resulted in a failure.
1.  .so files for native libraries were not included
2.  irregular version numbering scheme

These will be addressed in the version 5 release.

**Comment by HashEngineering** [comment]
We are working on this again with the goal to finish by August.  We have made the necessary changes to our source code and now are trying to build it on F-Droid before submitting it to their system.

**Comment by herbsmn** [comment]
thanks for the update and for the work you are doing!

**Comment by HashEngineering** [comment]
We are stuck with a problem, nearly similar to the problem we had in May.  We need support from F-Droid, but they haven't responded to our issue.

https://gitlab.com/fdroid/fdroiddata/issues/725

Perhaps others can help us.

**Comment by IzzySoft** [comment]
Status changed more than half a year back. For 8 months F-Droid waits on your response in the linked issue 😲 So where did this get stuck, and what should be the next steps?

Meanwhile, since 10/2016 the app is available [via my repo](https://apt.izzysoft.de/fdroid/index/apk/hashengineering.darkcoin.wallet) which can be added to the F-Droid client. Just in case this takes a little longer :wink:

**Comment by HashEngineering** [comment]
In my process, this got stuck in the build phase when I was verifying my meta data and build steps.

The APK generated by the fdroidserver didn't match that which was compiled by Android Studio that is posted to GitHub releases.  Within the APK, there were some resource files with different sizes.

**Comment by IzzySoft** [comment]
Thanks – looking forward then to getting it "unstuck" :wink:

**Comment by HashEngineering** [comment]
I made a lot of progress.  Building works, but it doesn't match the APK I have on GitHub, which is the problem I had 2-3 years ago.  I don't have this problem on my other apps that I have published on Fdroid.

**Comment by IzzySoft** [comment]
"doesn't match the APK" is not crucial (unless you want reproducible builds).

**Comment by HashEngineering** [comment]
I thought "Reproducable builds" builds was crucial.  I took the latest version, 6.6.4, and removed the GMS, etc.  And most of the non-matching files are now matching.

The APK's match against what I build on my linux machine except for the native libraries.  It seems that other people have similar issues of reproducible builds with NDK and gradle.  

```
...
ERROR: SHA-1 digest of lib/x86_64/libscrypt.so does not match the digest specified in META-INF/MANIFEST.MF. Expected: <FdycVxxGh4Hj1PDrG/JJzhgqIt4=>, actual: <Bf723JizEG7XV3mljU8KCg2Zxro=>
ERROR: SHA-1 digest of lib/x86_64/libx11.so does not match the digest specified in META-INF/MANIFEST.MF. Expected: <feIbxvXAj8X0bSrEYwJQgn0NcB8=>, actual: <aEkrwnQLbIWzpvY7ElLKEQa6U6w=>
...

DEBUG: > diff -r /tmp/tmpk0sya_2s/unsigned_binaries_hashengineering.darkcoin.wallet_60640.binary /tmp/tmpk0sya_2s/_tmp_tmpk0sya_2s_sigcp_hashengineering.darkcoin.wallet_60640
DEBUG: > /usr/bin/meld /tmp/tmpk0sya_2s/unsigned_binaries_hashengineering.darkcoin.wallet_60640.binary /tmp/tmpk0sya_2s/_tmp_tmpk0sya_2s_sigcp_hashengineering.darkcoin.wallet_60640
DEBUG: removing unsigned/hashengineering.darkcoin.wallet_60640.apk
DEBUG: removing unsigned/binaries/hashengineering.darkcoin.wallet_60640.binary.apk
ERROR: Could not build app hashengineering.darkcoin.wallet: compared built binary to supplied reference binary but failed
==== detail begin ====
Unexpected diff output:
==== detail end ====
INFO: Finished
INFO: 1 build failed

```

There is disagreement between the different means of comparing a diff tool and looking at the logs.  Perhaps I should submit it to Fdroid and see what results come up.  Perhaps they will know what to do.

**Comment by IzzySoft** [comment]
> I thought "Reproducable builds" builds was crucial.

It's really nice-to-have and a long-term goal, yes. But definitely not crucial. If you'd ask me for a single app out of the 2k+ at F-Droid building reproducible, I had to think (well, one I would be able to name – but for a second one I had to dig). So no, it's not "crucial" – but good if it's there. The advantage of course is F-Droid would ship the APK signed by you, so cross-updates are possible. If it does *not* build reproducible, F-Droid would ship the APK signed by F-Droid.

And yes, some of our packagers should be able to assist you – this is out of my area of proficiency (I'm neither a packager nor an Android dev).

As for what's crucial: having a build that comes without any proprietary libs, and without any blobs (JARs etc). This is what our bot checks when you open an RFP. If the bot says all's fine, you could go ahead submitting a merge request yourself (mentioning `closes: rfp#<issueNumber>` to have your RFP closed on merge).

Another really nice to have is [Fastlane file structure](https://gitlab.com/snippets/1895688) in your repo here, to provide description, screenshots etc. Big advantage: you'd need no MR at F-Droid to get that updated, as we'd grab the latest Fastlane details with the next tag automatically :wink:

PS, disclosure: I'm one of the F-Droid maintainers.

**Comment by HashEngineering** [comment]
@IzzySoft Thank you for this answer.

I will keep this issue updated on the status of my progress.

**Comment by HashEngineering** [comment]
Fdroid merge request submitted:
https://gitlab.com/fdroid/fdroiddata/merge_requests/6200

@IzzySoft, thank you for your help.

**Comment by IzzySoft** [comment]
Thanks! If you secure that one maven repo (using `https:` instead of `http:`), lint should succeed as well :wink:

**Comment by PanderMusubi** [comment]
Can https://github.com/HashEngineeringSolutions/dash-wallet get an upgrade from 7.0.9 to 7.2.6 so that version can be used by F-Droid users?

**Comment by HashEngineering** [comment]
I will try to get the changes made this week fo support this upgrade.

**Comment by HashEngineering** [comment]
A new tag was created to match today's release for v7.3.1.

https://github.com/HashEngineeringSolutions/dash-wallet/releases/tag/v7.3.1-foss

It may take some time before this is available on FDroid.  I would guess I will have to fix some build errors first.

**Comment by HashEngineering** [comment]
https://f-droid.org/wiki/page/hashengineering.darkcoin.wallet/lastbuild_70310

According to this, the build for Fdroid was successful today.

**Comment by HashEngineering** [comment]
Since the F-droid build has been updated, this issue will be closed.

**Comment by proofy** [comment]
Since Version 7.3.2 (70311) from 2021-08-21 
no newer version was deployed .
https://f-droid.org/de/packages/hashengineering.darkcoin.wallet/

Would be nice to reactivate the f-droid process.

Reopen this issue please
